src/components/generic-elements/download-button/download-button.js

```diff
@@ -1,5 +1,31 @@
 import './download-button.scss'
+import { createRef, useState } from 'react'
+import React from 'react'
+import {AuthenticatedHTTPClient} from '../../../utils/http-clients'
 
-export default (props) => (
-  <a className='download-button' href={props.url} target='_blank' rel='noopener noreferrer'>Download</a>
-)
+export function DownloadButton ({ url, format, filename }) {
+
+  const link = createRef()
+  const [downloadUrl, setDownloadUrl] = useState()
+  
+  React.useEffect(() => {
+    if (downloadUrl) {
+      link.current.click()
```

Comment:
I don't think I understand the purpose of this line.

Reply:
The flow of this component is kind of crazy.
When the user clicks the link, we call a presign url to get a key. (The key is based on the user's Auth0 info). When we get the key back (async), we update the link's url to point at the download endpoint with the proper key. We then click ourselves (re-click the link). Now when the link is clicked it starts the file download.

It might be more clear if we had a second (hidden) link, instead of updating the existing one. We need to 'manually' click the link so that we can trigger the chrome download functionality. At least that's my understanding.

I'm open to adding a comment or a second hidden link if that would make things more clear.

Reply:
I guess one, perhaps stupid, question is: can we just generate a presign url for every time the page is visited instead of only when the button is clicked?

Reply:
See my other comment. I think that would make things a lot better, but wouldn't fit the url quick expiration... =/